include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2344,6 +2594,8 @@ class binary_reader
     @note This function needs to respect the system's endianness, because
           bytes in CBOR, MessagePack, and UBJSON are stored in network order
           (big endian) and therefore need reordering on little endian systems.
+          On the other hand, BSON and BJData use little endian and should reorder
+          on big endian systems.
```

Comment:
Might be better to always use the `format` argument instead of sometimes using `format` and sometimes the template argument, and them sometimes not agreeing.  @nlohmann thoughts?

Reply:
I agree. The only benefit of using `InputIsLittleEndian` in the template is the ability to manually overwrite the input endianness regardless of the format (except bjdata) - maybe for testing purposes?

Currently, `InputIsLittleEndian` is only set to true in BSON, which seems to conform with the format requirement.